src/gateway/handlers/__tests__/make-render-handler.test.js

```diff
@@ -155,15 +167,111 @@ describe("#makeRenderHandler", () => {
             await handler(fakeRequest, fakeResponse);
 
             // Assert
-            expect(fakeRenderFn).toHaveBeenCalledWith(
+            expect(fakeRenderEnvironment.render).toHaveBeenCalledWith(
                 "THE_URL",
-                expect.objectContaining({
+                {
+                    logger: fakeLogger,
                     getHeader: expect.any(Function),
                     trace: expect.any(Function),
-                }),
+                },
             );
         });
 
+        describe("provided render API", () => {
```

Comment:
Filled in some testing gaps with tests that will be useful even after we implement proper header tracking (also fills in that 0.3% deficit in our coverage).